test/powershell/Modules/Microsoft.PowerShell.Core/Enter-PSHostProcess.Tests.ps1

```diff
@@ -34,32 +34,26 @@ function Invoke-PSHostProcessScript {
         [int] $Retry = 5 # Default retry of 5 times
     )
 
-    $sb = {
-        # use $i as an incrementally growing pause based on the attempt number
-        # so that it's more likely to succeed.
-        $commandStr = @'
-Start-Sleep -Seconds {0}
-Enter-PSHostProcess {1} -ErrorAction Stop
+    $commandStr = @'
+Enter-PSHostProcess {0} -ErrorAction Stop
 $PID
 Exit-PSHostProcess
-'@ -f $i, $ArgumentString
-
-        ($commandStr | & $powershell -c -) -eq $Id
-    }
+'@ -f $ArgumentString
 
     $result = $false
-    $failures = 0
     foreach ($i in 1..$Retry) {
-        if ($sb.Invoke()) {
-            $result = $true
+        # use $i as an incrementally growing pause based on the attempt number
+        # so that it's more likely to succeed.
+        Start-Sleep -Seconds $i
+
+        $result = ($commandStr | & $powershell -noprofile -c -) -eq $Id
+        if ($result) {
             break
         }
-
-        $failures++
     }
 
-    if($failures) {
-        Write-Warning "Enter-PSHostProcess script failed $i out of $Retry times."
+    if ($i -gt 1) {
+        Write-Verbose -Verbose "Enter-PSHostProcess script failed $i out of $Retry times."
```

Comment:
It would be nice to know if this failed because the target process has terminated or its named pipe path/listener failed.  It is easy to check for termination by trying to get the process by Id:
```powershell
$proc = Get-Process -Id $Id -ErrorAction Ignore
$hasExited = ($proc -eq $null) -or $proc.HasExited
```

If the process terminated somehow, then we can break out of the loop immediately.

Reply:
Added some code for diagnosing